gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/GradebookNgBusinessService.java

```diff
@@ -683,20 +684,19 @@ public GradeSaveResponse saveGrade(final Long assignmentId, final String student
 		// this gives us our base list and will be sorted as per our desired
 		// sort method
 		final List<User> students = getUsers(studentUuids);
-		if (settings.getStudentSortOrder() != null || settings.getNameSortOrder() != null) {
```

Comment:
settings.getNameSortOrder is never null, causing this block to execute and sort users by name every time